### PR TITLE
Sky130

### PR DESCRIPTION
**Pull request by nayiri-k**
1) add endcaps to PAR
2) connect global nets VDD/VSS to the pg_pins in the sky130 standard cells (VPWR, VGND, VPB, VNB)
3) other minor edits to plugin

### CODE REVIEW

File: src/hammer-vlsi/technology/sky130/__init__.py [hunk 1]
```diff
@@ -264,5 +269,30 @@ def sky130_power_nets(ht: HammerTool) -> bool:
                 ht.verbose_append("connect_global_net {tie} -type net -net_base_name {net}".format(tie=pwr_gnd_net.tie, net=pwr_gnd_net.name))
     return True
 
-tech = SKY130Tech()
+def sky130_connect_nets(ht: HammerTool) -> bool:
```
Comment (harrisonliew) [inline src/hammer-vlsi/technology/sky130/__init__.py:272]: Hm, this really should not need to be a hook. Question: are you using a CPF, or allowing Hammer to auto-generate the power intent? If it's the former, these global net connections should be in the CPF, If it's the latter, we really should add this `connect_global_net` here instead: https://github.com/ucb-bar/hammer-cadence-plugins/blob/4ecbce344302a5e697c2d54fd2fd80c3f70db186/par/innovus/__init__.py#L298

**Review by harrisonliew — CHANGES_REQUESTED** [review]

File: src/hammer-vlsi/technology/sky130/__init__.py (hunk 1 above)
Comment (harrisonliew) [reply src/hammer-vlsi/technology/sky130/__init__.py:272]: Also, I think the DRC & LVS hooks should be made default (comments in ucb-bar/chipyard#1115).

File: src/hammer-vlsi/technology/sky130/__init__.py [hunk 2]
```diff
@@ -264,5 +269,30 @@ def sky130_power_nets(ht: HammerTool) -> bool:
...
+    assert isinstance(
```
Comment (harrisonliew) [inline src/hammer-vlsi/technology/sky130/__init__.py:273]: You also need an assertion that `ht` is a `TCLTool` to get rid of the type checker failures against `append` and `verbose_append`.

**Review by harrisonliew — CHANGES_REQUESTED** [review]

File: src/hammer-vlsi/technology/sky130/__init__.py (hunk 1 above)
Comment (nayiri-k) [reply src/hammer-vlsi/technology/sky130/__init__.py:272]: Made DRC/LVS hooks default now, and will move the sky130_power_nets hook to the auto-generated CPF in a future PR.

File: src/hammer-vlsi/technology/sky130/defaults.yml [hunk 3]
```diff
@@ -18,8 +18,9 @@ technology.sky130:
 
 # Mentor environment variables
 # Override this in project
+mentor.extra_env_vars_meta: lazydeepsubst
 mentor.extra_env_vars:
-  - PDK_HOME: "<path-to-skywater-src-nda>/s8/<version#>"
+  - PDK_HOME: "${technology.sky130.sky130_nda}/s8/V2.0.1"
```
Comment (harrisonliew) [inline src/hammer-vlsi/technology/sky130/defaults.yml:23]: Oh, turns out that this is exactly the same as `technology.sky130.pdk_home`

Comment (harrisonliew) [reply src/hammer-vlsi/technology/sky130/defaults.yml:23]: Nvm, you can't lazy twice.

**Review by harrisonliew — APPROVED** [review]

**Pull request merged**